api_v3/services/PartnerService.php

```diff
@@ -16,79 +16,118 @@ protected function partnerRequired($actionName)
+			return true;
+		}
+		catch ( SignupException $se )
+		{
+			throw new KalturaAPIException( APIErrors::PARTNER_REGISTRATION_ERROR, $se->getMessage());
+		}
+		catch ( Exception $ex )
+		{
+			KalturaLog::CRIT($ex);
+			// this assumes the partner name is unique - TODO - remove key from DB !
+			throw new KalturaAPIException( APIErrors::PARTNER_REGISTRATION_ERROR, 'Unknown error');
+		}
+	}
+	
+	protected function getIgnorePassword($existingUser, $isAdminOrVarConsole, $adminEmail, $parentPartnerId)
+	{
+		$ignorePassword = false;
+		if ($existingUser && $isAdminOrVarConsole)
+		{
+			kuserPeer ::setUseCriteriaFilter(false);
+			$kuserOfLoginData = kuserPeer ::getKuserByEmail($adminEmail,
+			                                                $existingUser -> getConfigPartnerId());
+			kuserPeer ::setUseCriteriaFilter(true);
+			if ($kuserOfLoginData &&
+				(!$parentPartnerId || ($parentPartnerId == $existingUser -> getConfigPartnerId())))
 			{
-				$parentPartnerId = $this->getKs()->partner_id;
-				if ($parentPartnerId == Partner::ADMIN_CONSOLE_PARTNER_ID) {
-		                    $parentPartnerId = null;
-		                    $isAdminOrVarConsole = true;
-				}
-                else
-                {
-					// only if this partner is a var/group, allow setting it as parent for the new created partner
-					$parentPartner = PartnerPeer::retrieveByPK( $parentPartnerId );
-					if ( ! ($parentPartner->getPartnerGroupType() == PartnerGroupType::VAR_GROUP ||
-							$parentPartner->getPartnerGroupType() == PartnerGroupType::GROUP ) )
-					{
-						throw new KalturaAPIException( KalturaErrors::NON_GROUP_PARTNER_ATTEMPTING_TO_ASSIGN_CHILD , $parentPartnerId );
-					}
-					$isAdminOrVarConsole = true;
-					if ($templatePartnerId)
-					{
-					    $templatePartner = PartnerPeer::retrieveByPK($templatePartnerId);
-					    if (!$templatePartner || $templatePartner->getPartnerParentId() != $parentPartnerId)
-					        throw new KalturaAPIException( KalturaErrors::NON_GROUP_PARTNER_ATTEMPTING_TO_ASSIGN_CHILD , $parentPartnerId );
-					}
-				}
+				$ignorePassword = true;
 			}
-			
+		}
+		return $ignorePassword;
+	}
+	
+	/**
+	 * Create a new Partner object
+	 * 
+	 * @action register
+	 * @param KalturaPartner $partner
+	 * @param string $cmsPassword
+	 * @param int $templatePartnerId
+	 * @param bool $silent
+	 * @return KalturaPartner
+	 * @ksOptional
+	 *
+	 * @throws APIErrors::PARTNER_REGISTRATION_ERROR
+	 */
+	public function registerAction( KalturaPartner $partner , $cmsPassword = "" , $templatePartnerId = null, $silent = false)
+	{
+		KalturaResponseCacher::disableCache();
+		
+		try
+		{
+			$c = new Criteria();
+			$c->addAnd(UserLoginDataPeer::LOGIN_EMAIL, $partner->adminEmail, Criteria::EQUAL);
+			$existingUser = UserLoginDataPeer::doSelectOne($c);
+			/* @var $exisitingUser UserLoginData */
+			$this->registrationValidationAction($partner, $cmsPassword, $templatePartnerId, $silent);
```

Comment:
Assumption is that the action is being called from the controller and we should avoid internal calling +  you can call $partner_registration->validateNewPartner here, all the inputs arguments are here anyway